client/Main.elm

```diff
@@ -59,9 +56,10 @@ type Msg
 
 
 type FromServer
-    = Initial (List ItemId)
-    | NewItem Item
-    | Delete ItemId
+    = Initial (Result Http.Error (List ItemId))
+    | NewItem (Result Http.Error Item)
+    | CreatedItem (Result Http.Error ItemId)
```

Comment:
Exactly. This was the major braking change where `Http.send` want's a type constructor as argument and the user is handling a result. There could be ways to reduce boilerplate or clutter if all error's should be handled the same way. The way it's implemented now is the most straight forward way. In the [guide](https://guide.elm-lang.org/architecture/effects/http.html) it's done the same using pattern matching instead of `case-of`.